backend/app/adapter/facebook/identityprovider_integration_test.go

```diff
@@ -0,0 +1,149 @@
+// +build integration all
+
+package facebook
+
+import (
+	"bytes"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"testing"
+
+	"github.com/short-d/app/mdtest"
+)
+
+func TestIdentityProvider_GetAuthorizationURL(t *testing.T) {
+	t.Parallel()
+	httpRequest := mdtest.NewHTTPRequestFake(
+		func(req *http.Request) (response *http.Response, e error) {
+			return nil, nil
+		})
+	clientID := "id_12345"
+	clientSecret := "client_secret"
+	redirectURI := "http://localhost/oauth/facebook/sign-in/callback"
+	identityProvider := NewIdentityProvider(httpRequest, clientID, clientSecret, redirectURI)
+
+	urlResponse := identityProvider.GetAuthorizationURL()
+
+	parsedUrl, err := url.Parse(urlResponse)
+	mdtest.Equal(t, nil, err)
+	mdtest.Equal(t, "https", parsedUrl.Scheme)
+	mdtest.Equal(t, "www.facebook.com", parsedUrl.Host)
+	mdtest.Equal(t, "/v4.0/dialog/oauth", parsedUrl.Path)
+	mdtest.Equal(t, "public_profile,email", parsedUrl.Query().Get("scope"))
```

Comment:
Scope should be parsed into slice in order to verify because we shouldn't assume the scope can only be arranged in this order. You can use this function to assert the content of an array: https://github.com/short-d/app/blob/master/mdtest/assert.go#L17

Reply:
good catch! will change accordingly